pup-0002.md

```diff
@@ -0,0 +1,97 @@
+Sphinx style docstrings.
+
+## Details
+
+Upon acceptance, these changes will be documented in the Pulp style guide.
+
+### Style Choices
+
+Docstrings should follow the conventions documented in the
+[Google style guide](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments)
+except where Pulp specific additions or exceptions apply.
+
+Additionally, docstrings must be parsable by the
+[Napoleon plugin for Sphinx.](http://www.sphinx-doc.org/en/stable/ext/napoleon.html)
+
+### Additions/Exceptions
+
+Pulp conventions that differ from the `Comments` section of the Google style guide will be
+explicitly included in the Pulp style guide.
+
+1. Modules will not contain license information. Our license information is presented correctly
+within the project; it is unnecessary to include it in each file.
+
+2. The type of each `Arg` value should be included after the variable name in parentheses. The type of each `Returns` value should be the first item on the line.
+
+3. Following the type of `Arg` and `Return` values, there will be a colon and a single space
+followed by the description. Additional spaces should not be used to align types and descriptions.
+
+4. `Fields` and `Relations` sections will be used when documenting fields on Django models.
+The `Fields` section will be used for non-related fields on Model classes, the `Relations`
+section will be used for related fields on Model classes.
+
+### Porting from Pulp 2.Y
+
+The 3.0 development guide will be updated to state that all ported code should contain updated
+docstrings. Regular expressions to ease large changes will be included there.
+
+## Drawbacks
+
+It will take time to convert each docstrings into the new style.
+
+## Alternatives
+
+### Use default Sphinx style docstrings
+
+We could use the default Sphinx docstrings that we used in the 2.Y line.
+
+Reasons to do this:
+  - We would not need to port old docstrings
+  
+Reasons not to do this:
+  - Porting docstrings is quick
+  - We already have mostly Google style docstrings in the new code
+
+### Leave existing old style docstrings for old code
+
+We could write new style docstrings for new code only, leaving existing old style docstrings
+unchanged.
+
+Reasons to do this:
+  - It would be less work
+
+Reasons not to do this:
+  - It is less readable to have multiple styles in the same code base
+  - It could be confusing to developers to know when they should or must update an old docstring to the new style
+
+### Tabular style
+
+The guidelines as written use a single space after the type of `Arg` and `Return` values. 
+A variation of Google style uses extra spaces to vertically align the types of objects.
+
+Example:
+
+    Args:
+        arg1                 (str): Lots of space to align with second arg
+        much_longer_argument (str): If I change this line, I have to change first line too
+
+
+Reasons to do this:
+  - Some developers find this more readable
```

Comment:
I don't necessarily find it more readable and question whether it's worth the time to do the formatting.  But don't feel strongly.